File: tasks/implementations/java/tasks/sentiment_analysis.java
import java.nio.file.*;
import java.util.*;

class SentimentAnalysis {
    static final Set<String> POSITIVE = Set.of("good", "great", "happy", "clean", "fast", "love");
    static final Set<String> NEGATIVE = Set.of("bad", "sad", "dirty", "slow", "hate", "poor");

    public static void run(String size, String fixturesRoot) throws Exception {
        long total = 0;
        for (String line : Files.readAllLines(Path.of(fixturesRoot, "generated", "sentiment", size + ".txt"))) {
            for (String token : line.trim().split("\\s+")) {
                if (POSITIVE.contains(token)) total += 1;
                else if (NEGATIVE.contains(token)) total -= 1;
            }
        }
        System.out.println(total);
    }
}

File: tasks/implementations/python2/tasks/sentiment_analysis.py
from tasks.compat import emit, join_path, read_lines

POSITIVE = set(['good', 'great', 'happy', 'clean', 'fast', 'love'])
NEGATIVE = set(['bad', 'sad', 'dirty', 'slow', 'hate', 'poor'])


def run(size, fixtures_root):
    path = join_path(fixtures_root, 'generated', 'sentiment', '%s.txt' % size)
    total = 0
    for line in read_lines(path):
        for token in line.split():
            if token in POSITIVE:
                total += 1
            elif token in NEGATIVE:
                total -= 1
    emit(total)
